util/log: TestHTTPSinkTimeout failed

util/log.TestHTTPSinkTimeout [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15899524?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15899524?buildTab=artifacts#/) on master @ [9fa95e40accdd4a16c80fb96604c6fb2bf56036a](https://github.com/cockroachdb/cockroach/commits/9fa95e40accdd4a16c80fb96604c6fb2bf56036a):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestHTTPSinkTimeout (4m7s)

```
Stack: 

```
goroutine 691 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestHTTPSinkTimeout
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/d7e801376927e0a075813f3d68bf5bd0/logTestHTTPSinkTimeout982601034
    http_sink_test.go:128: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/log/http_sink_test.go:128
        	            				github.com/cockroachdb/cockroach/pkg/util/log/http_sink_test.go:207
        	Error:      	"1.003317757s" is not less than or equal to "1s"
        	Test:       	TestHTTPSinkTimeout
        	Messages:   	Log call exceeded timeout, expected to be less than 1s, got 1.003317757s
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHTTPSinkTimeout.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39966

util/log.TestHTTPSinkTimeout [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16025508?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16025508?buildTab=artifacts#/) on master @ [1792002e50ad8eea4c80500b26f9d243817fcfd1](https://github.com/cockroachdb/cockroach/commits/1792002e50ad8eea4c80500b26f9d243817fcfd1):

Fatal error:

```
panic: test timed out after 4m57s
running tests:
	TestHTTPSinkTimeout (4m14s)

```
Stack: 

```
goroutine 649 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestHTTPSinkTimeout
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/d7e801376927e0a075813f3d68bf5bd0/logTestHTTPSinkTimeout1447145162
    http_sink_test.go:128: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/util/log/http_sink_test.go:128
        	            				github.com/cockroachdb/cockroach/pkg/util/log/http_sink_test.go:207
        	Error:      	"1.036815461s" is not less than or equal to "1s"
        	Test:       	TestHTTPSinkTimeout
        	Messages:   	Log call exceeded timeout, expected to be less than 1s, got 1.036815461s
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHTTPSinkTimeout.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/util/log/http_sink_test.go
+++ b/pkg/util/log/http_sink_test.go
@@ -121,18 +121,15 @@ func testBase(
 	Ops.Infof(context.Background(), "hello world")
 	logDuration := timeutil.Since(logStart)
 
-	// Note: deadline is passed by the caller and already contains slack
-	// to accommodate for the overhead of the logging call compared to
-	// the timeout in the HTTP request.
-	if deadline > 0 && logDuration > deadline {
+	if deadline > 0 {
+		// Note: deadline is passed by the caller and already contains slack
+		// to accommodate for the overhead of the logging call compared to
+		// the timeout in the HTTP request.
 		require.LessOrEqualf(t, logDuration, deadline,
 			"Log call exceeded timeout, expected to be less than %s, got %s", deadline.String(), logDuration.String())
-	}
-
-	// If we don't properly hang in the handler when we want to test a
-	// timeout, we'll just log very quickly. This check ensures that we
-	// catch that testing error.
-	if deadline > 0 && logDuration < *defaults.Timeout {
+		// If we don't properly hang in the handler when we want to test a
+		// timeout, we'll just log very quickly. This check ensures that we
+		// catch that testing error.
 		require.Greaterf(t, logDuration, *defaults.Timeout,
 			"Log call was too fast, expected to be greater than %s, got %s", defaults.Timeout.String(), logDuration.String())
 	}
@@ -204,7 +201,7 @@ func TestHTTPSinkTimeout(t *testing.T) {
 		},
 	}
 
-	testBase(t, defaults, nil /* testFn */, true /* hangServer */, 1*time.Second, time.Duration(0))
+	testBase(t, defaults, nil /* testFn */, true /* hangServer */, 10*time.Second, time.Duration(0))
 }
 
 // TestHTTPSinkContentTypeJSON verifies that the HTTP sink content type

